### PR TITLE
Citation Plugin v1.0.0

### PR DESCRIPTION
**Pull request by sam1360**

### CODE REVIEW

File: Plugin/citation-plugin.html [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<!DOCTYPE html>
```
Comment (nokeeo) [inline Plugin/citation-plugin.html:1]: Remove dead code in the entire file.

File: Plugin/package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "citation-plugin",
+  "version": "0.0.1",
```
Comment (nokeeo) [inline Plugin/package.json]: Change version to 1.0.0

File: Plugin/.eslintignore [hunk 3]
```diff
@@ -0,0 +1,6 @@
+
+/node_modules/*
+**/*.html
+**/*.md
+**/*.json
+**/*.conf
```
Comment (nokeeo) [inline Plugin/.eslintignore]: New line to end file

File: Plugin/citation-plugin.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
+const CitationCore = require('citation-core');
+
+const resultNode = document.getElementById('result');
+const citationFormatsNode = document.getElementById('citationFormats');
+const textNode = document.getElementById('citation-URL');
+
```
Comment (nokeeo) [inline Plugin/citation-plugin.js:6]: Please add JSdoc comments to all functions.

File: Plugin/style.css [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+.outputTextArea {
+    height:160px;
+    width:350px;
+    resize:none;
+}
+
+input[type="number"] {
+    background-color: #fff;
+    border: 1px solid #b1b1b1;
+    box-shadow: 0 0 0 0 rgba(97, 181, 255, 0);
```
Comment (nokeeo) [inline Plugin/style.css]: rgba not using alpha.  Use hex instead

File: Plugin/.eslintrc.json [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+        "max-len": "off",
+        "brace-style": ["error", "stroustrup"],
+        "no-unused-vars": ["error", { "args": "none" }]
+    },
+    "env": {
+        "browser": true,
+        "node": true,
+        "webextensions": true
+    }
+}
```
Comment (nokeeo) [inline Plugin/.eslintrc.json]: No new line

File: Plugin/.eslintrc.json [hunk 7]
```diff
@@ -0,0 +1,16 @@
+{
+    "extends": "airbnb-base",
+    "plugins": [
+        "import"
+    ],
+    "rules": {
+        "max-len": "off",
+        "brace-style": ["error", "stroustrup"],
+        "no-unused-vars": ["error", { "args": "none" }]
```
Comment (nokeeo) [inline Plugin/.eslintrc.json]: Just a note we added a rule in [this](https://github.com/mozillascience/CitationCore/pull/6) PR.  Not merged, but we might want to add that rule here.

File: Plugin/package.json [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/mozillascience/citation-plugin.git"
+  },
+  "keywords": [
+    "citation"
+  ],
+  "author": "Samuel Mosher",
```
Comment (nokeeo) [inline Plugin/package.json]: Please add authors of the project and their emails. Should also include the mozilla user. An example can be found [here](https://github.com/mozillascience/CitationCore/blob/master/package.json).

File: README.md [hunk 9]
```diff
@@ -1 +1,19 @@
...
+npm install 
+```
+
+Once the install script finishes, navigate to [about:debugging#addons](about:debugging#addons). Once there, select the the "Load Temporary Add-on", navigate to the Plugin directory and select the manifest.json and click load
+## Contributor Install
+If you are interested in contributing to citation-plugin please follow these install instructions. They will add a pre-commit hook that will run our linter and reject commits that do not meet the project's coding standards.  We are adhereing to the [AirBnb style guide](https://github.com/airbnb/javascript). 
+```
+git clone https://github.com/mozillascience/citation-plugin.git
+cd Plugin
+cp dev/pre-commit .git/hooks/
```
Comment (nokeeo) [inline README.md:18]: Please include this hook script in the PR.

File: Plugin/.eslintignore [hunk 10]
```diff
@@ -0,0 +1,6 @@
+
+/node_modules/*
+**/*.html
```
Comment (nokeeo) [inline Plugin/.eslintignore]: Just curious, why are we ignoring files that match these patterns

**Review by nokeeo — CHANGES_REQUESTED** [review]

File: Plugin/citation-plugin.html [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+    <head>
+        <meta charset="utf-8">
+        <link rel="stylesheet" href="style.css">
+    </head>
+    <body class="panel">
+        <form id="citation-form">
+                <table>
+                  <tr>
+                    <td><label for="citation-URL">Search URL</label></td>
+                    <td><input type="text" style="width:100%;" id="citation-URL"></td>
```
Comment (faokryn) [inline Plugin/citation-plugin.html]: Why are you using the style class instead of a class?
(Also below)

File: Plugin/manifest.json [hunk 12]
```diff
@@ -0,0 +1,19 @@
+{
+    "manifest_version": 2,
+    "name": "Software Citation Tools Plugin",
+    "version": "1.0.0",
+    "description": "A demo for the Software Citation Tools Plugin",
```
Comment (faokryn) [inline Plugin/manifest.json]: What is the purpose of manifest.json? Should this be referred to as a demo?

File: Plugin/style.css [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+.outputTextArea {
+    height:160px;
+    width:350px;
+    resize:none;
+}
+
+input[type="number"] {
+    background-color: #fff;
```
Comment (faokryn) [inline Plugin/style.css]: Keep the color style consistent. Use #ffffff

File: README.md [hunk 14]
```diff
@@ -1 +1,19 @@
...
+citation-plugin is a firefox plugin which presents the [CitationCore](https://github.com/mozillascience/CitationCore) library in a user friendly way in browswer. It is part of a larger effort, lead by Mozilla Science Lab, to develop a suite of tools that aim to implement a standard for citing software and making it easier to cite software correctly.  To learn more about this project you can visit the [Software Citation Tools repository](https://github.com/mozillascience/software-citation-tools).
+
+## Install
+To install:
+```
+cd Plugin
+npm install 
+```
+
+Once the install script finishes, navigate to [about:debugging#addons](about:debugging#addons). Once there, select the the "Load Temporary Add-on", navigate to the Plugin directory and select the manifest.json and click load
```
Comment (faokryn) [inline README.md]: In 'select the the "Load Temporary Add-On"', "the" is repeated twice. However, it isn't followed up by "button" or "link" or the like, and the article should probably be removed all together, changing it to 'select "Load Temporary Add-On"'

File: dev/pre-commit [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+  fi
+done
+
+if $passing; then
+  echo "Linter sucessfully passed. Commiting..."
+  exit 0
+else
+  echo "Linter failed. Please correct errors. Aborting commit..."
+  exit 1
+fi
```
Comment (faokryn) [inline dev/pre-commit]: Add EOF new line.

File: Plugin/.eslintignore [hunk 16]
```diff
@@ -0,0 +1,5 @@
+
+/node_modules/*
+**/*.md
+**/*.json
+**/*.conf
```
Comment (faokryn) [inline Plugin/.eslintignore]: Add EOF new line.

**Review by faokryn — CHANGES_REQUESTED** [review]

File: README.md (hunk 14 above)
Comment (faokryn) [reply README.md]: Any update on this requested change?

File: dev/pre-commit (hunk 15 above)
Comment (faokryn) [reply dev/pre-commit]: Any update on this requested change?

**Review by faokryn — APPROVED** [review]

**Comment by faokryn** [comment]
👍 lgtm assuming Eric's satisfied with the JSdoc comments and hook script

**Review by nokeeo — APPROVED** [review]

**Pull request merged**